doc/ControlProtocol/ControlProtocol.tex

```diff
@@ -0,0 +1,886 @@
+        m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|m{0.04cm}|}
+        \hline
+        Word & 
+        \begin{sideways}31\end{sideways} &
+        \begin{sideways}30\end{sideways} & 
+        \begin{sideways}29\end{sideways} &
+        \begin{sideways}28\end{sideways} &
+        \begin{sideways}27\end{sideways} &
+        \begin{sideways}26\end{sideways} &
+        \begin{sideways}25\end{sideways} &
+        \begin{sideways}24\end{sideways} &
+        \begin{sideways}23\end{sideways} &
+        \begin{sideways}22\end{sideways} &
+        \begin{sideways}21\end{sideways} &
+        \begin{sideways}20\end{sideways} &
+        \begin{sideways}19\end{sideways} &
+        \begin{sideways}18\end{sideways} &
+        \begin{sideways}17\end{sideways} &
+        \begin{sideways}16\end{sideways} &
+        \begin{sideways}15\end{sideways} &
+        \begin{sideways}14\end{sideways} &
+        \begin{sideways}13\end{sideways} &
+        \begin{sideways}12\end{sideways} &
+        \begin{sideways}11\end{sideways} &
+        \begin{sideways}10\end{sideways} &
+        \begin{sideways}9\end{sideways} &
+        \begin{sideways}8\end{sideways} &
+        \begin{sideways}7\end{sideways} &
+        \begin{sideways}6\end{sideways} &
+        \begin{sideways}5\end{sideways} &
+        \begin{sideways}4\end{sideways} &
+        \begin{sideways}3\end{sideways} &
+        \begin{sideways}2\end{sideways} &
+        \begin{sideways}1\end{sideways} &
+        \begin{sideways}0\end{sideways} \\
+        \hline
+        [0,1] & \multicolumn{32}{c|}{Header}\\
+        \hline
+        2 & \multicolumn{32}{c|}{Git Commit Hash}\\
+        \hline
+        3 & \multicolumn{32}{c|}{Robot UUID}\\
+        \hline
+        4 & \multicolumn{8}{c|}{Robot State} & \multicolumn{8}{c|}{\cellcolor[gray]{.6}Reserved}&
+            \multicolumn{16}{c|}{Number of config Entries}\\
+        \hline
+        ... & \multicolumn{32}{c|}{Arbitrary Data}\\
+        \hline
+        n & \multicolumn{16}{c|}{Arbitrary Data (Cont)} & \multicolumn{16}{c|}{\cellcolor[gray]{.6}Unused}\\
+        \hline
+    \end{tabular}
+\end{table}
+Git Commit Hash: The hash of the current git commit of the FMS or Robot code (not core code) running on the sender.
+The hash should be inserted into the code at compile time (i.e. using a makefile)\\
+Robot UUID: A unique identifier for the robot which MAY NOT conflict with any other robots on the field.\\
+Robot State: The current state of the robot (see the *\_STATE message definitions for a more complete description)\\
+Number of Config Entries: The current number of configuration entries the robot has\\
+Arbitrary Data: Whatever data the programmer sees fit to send in a debug message to the reciever.\\
+Unused:  Unused padding bits to align the message to a 32 bit word.\\
+
+\clearpage
+
+\section{Failure Modes}
```

Comment:
Maybe it would be a good idea to put a section for example interactions with diagrams

Reply:
I agree that it would be, but given our tight schedule, it may not be the place to spend time right now.